internal/app/triggermessageprocessor.go

```diff
@@ -0,0 +1,143 @@
+func (mp *triggerMessageProcessor) Process(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope) error {
+	context, ok := ctx.(*appfunction.Context)
+	if !ok {
+		return fmt.Errorf("App Context must be an instance of internal appfunction.Context. Use NewAppContext to create instance.")
+	}
+
+	defaultPipelines := mp.bnd.GetMatchingPipelines(interfaces.DefaultPipelineId)
+
+	if len(defaultPipelines) != 1 {
+		return fmt.Errorf("TriggerMessageProcessor does not support multiple pipelines.  Please use TriggerMessageHandler for multi-pipeline support")
+	}
+
+	messageError := mp.bnd.ProcessMessage(context, envelope, defaultPipelines[0])
+	if messageError != nil {
+		// ProcessMessage logs the error, so no need to log it here.
+		return messageError.Err
+	}
+
+	return nil
+}
+
+// MessageReceived provides runtime orchestration to pass the envelope / context to configured pipeline(s).
+func (mp *triggerMessageProcessor) MessageReceived(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope, outputHandler interfaces.PipelineResponseHandler) error {
+	return mp.messageReceived(ctx, envelope, outputHandler)
+}
+
+func (mp *triggerMessageProcessor) messageReceived(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope, responseHandler interfaces.PipelineResponseHandler) error {
+	lc := mp.bnd.LoggingClient()
+
+	lc.Tracef("custom trigger attempting to find pipeline(s) for topic %s", envelope.ReceivedTopic)
+
+	// ensure we have a context established that we can safely cast to *appfunction.Context to pass to runtime
+	if _, ok := ctx.(*appfunction.Context); ctx == nil || !ok {
+		ctx = mp.bnd.BuildContext(envelope)
+	}
+
+	pipelines := mp.bnd.GetMatchingPipelines(envelope.ReceivedTopic)
+
+	lc.Tracef("trigger found %d pipeline(s) that match the incoming topic '%s'", len(pipelines), envelope.ReceivedTopic)
+
+	if responseHandler == nil && len(pipelines) != 0 {
+		return fmt.Errorf("must specify a response responseHandler when multiple pipelines are configured")
+	}
+
+	var finalErr error
+
+	pipelinesWaitGroup := sync.WaitGroup{}
+
+	for _, pipeline := range pipelines {
+		pipelinesWaitGroup.Add(1)
+		go func(p *interfaces.FunctionPipeline, wg *sync.WaitGroup, errCollector func(error)) {
+			defer wg.Done()
+
+			lc.Tracef("trigger sending message to pipeline %s (%s)", p.Id, envelope.CorrelationID)
+
+			if msgErr := mp.bnd.ProcessMessage(ctx.Clone().(*appfunction.Context), envelope, p); msgErr != nil {
+				lc.Errorf("message error in pipeline %s (%s) %s", p.Id, envelope.CorrelationID, msgErr.Err.Error())
+				errCollector(msgErr.Err)
+			} else {
+				if responseHandler != nil {
+					if outputErr := responseHandler(ctx, p); outputErr != nil {
+						lc.Errorf("failed to process output for message %s on pipeline %s: %s", ctx.CorrelationID(), p.Id, outputErr.Error())
```

Comment:
In case correlation ID is empty. Nothing enforces that it is set... 
```suggestion
						lc.Errorf("failed to process output for message '%s' on pipeline %s: %s", ctx.CorrelationID(), p.Id, outputErr.Error())
```